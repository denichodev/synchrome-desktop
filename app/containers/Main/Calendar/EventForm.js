import React, { Component } from 'react';
import { Field, reduxForm } from 'redux-form';

class EventForm extends Component {
  renderTextField = field => {
    // const { meta: { touched, error } } = field;
    // const className = `form-group ${touched && error ? 'has-danger' : ''}`;
    const { label, name, input, placeholder } = field;
    return (
      <div className="form-group">
        <label htmlFor={name}>
          {label}
        </label>
        <input
          type="text"
          className="form-control"
          placeholder={placeholder}
          {...input}
        />
      </div>
    );
  };

  renderForm = () => (
    <form>
      <Field
        label="Event Name"
        name="name"
        placeholder="Enter your event name here"
        component={this.renderTextField}
      />
      <Field
        label="Start date"
        name="start"
        placeholder="Enter start date"
        component={this.renderTextField}
      />
      <Field
        label="End date"
        name="end"
        placeholder="Enter end date"
        component={this.renderTextField}
      />
    </form>
  );

  render() {
    return (
      <div className="animated fadeIn">
        <div className="row">
          <div className="col-md-12 col-sm-12">
            <div className="card">
              <div className="card-header">Add New Event</div>
              <div className="card-block">
                {this.renderForm()}
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

const formOptions = {
  form: 'EventsNewForm',
  destroyOnUnmount: false
};

export default reduxForm(formOptions)(EventForm);
