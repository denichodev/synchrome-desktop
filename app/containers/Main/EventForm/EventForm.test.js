import React from 'react';
import EventForm from './EventForm';

function setup() {
  const props = {};

  const enzymeWrapper = shallow(<EventForm {...props} />);

  return {
    props,
    enzymeWrapper
  };
}

describe('EventForm', () => {
  it('should match snapshot', () => {
    const { enzymeWrapper } = setup();
    expect(enzymeWrapper).toMatchSnapshot();
  });
});
