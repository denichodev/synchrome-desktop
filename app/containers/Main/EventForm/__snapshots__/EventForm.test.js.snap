// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventForm should match snapshot 1`] = `
<Connect(Form(EventForm))
  destroyOnUnmount={true}
  enableReinitialize={false}
  forceUnregisterOnUnmount={false}
  form="EventsNewForm"
  getFormState={[Function]}
  keepDirtyOnReinitialize={false}
  persistentSubmitErrors={false}
  pure={true}
  shouldAsyncValidate={[Function]}
  shouldValidate={[Function]}
  touchOnBlur={true}
  touchOnChange={false}
/>
`;
